state/relation.go

```diff
@@ -332,6 +332,43 @@ func (r *Relation) Destroy() error {
 	return err
 }
 
+// destroyCrossModelRelationUnitsOps returns the operations necessary for the units
+// of the specified remote application to leave scope.
+func destroyCrossModelRelationUnitsOps(op *ForcedOperation, remoteApp *RemoteApplication, rel *Relation, onlyForTerminated bool) ([]txn.Op, error) {
+	if onlyForTerminated {
+		statusInfo, err := remoteApp.Status()
+		if op.FatalError(err) && !errors.IsNotFound(err) {
+			return nil, errors.Trace(err)
+		}
+		if err != nil || statusInfo.Status != status.Terminated {
```

Comment:
I don't get this, can we add some comments on why this is sound i.e. it's ok to `return nil, nil` for an error?

Reply:
this should return `ErrNoOperations`

Reply:
fixed in https://github.com/juju/juju/pull/13353/commits/2f1d2fb70182dbce2d8fe9d48f6f8e8b0641a440